src/OpenTelemetry.AutoInstrumentation/Configuration/MeterSettings.cs

```diff
@@ -36,41 +37,11 @@ private MeterSettings(IConfigurationSource source)
             MetricExporter = ParseMetricExporter(source);
             ConsoleExporterEnabled = source.GetBool(ConfigurationKeys.Metrics.ConsoleExporterEnabled) ?? false;
 
-            var instrumentations = new Dictionary<string, MeterInstrumentation>();
-            var enabledInstrumentations = source.GetString(ConfigurationKeys.Metrics.Instrumentations);
-            if (enabledInstrumentations != null)
-            {
-                foreach (var instrumentation in enabledInstrumentations.Split(Separator))
-                {
-                    if (Enum.TryParse(instrumentation, out MeterInstrumentation parsedType))
-                    {
-                        instrumentations[instrumentation] = parsedType;
-                    }
-                    else
-                    {
-                        throw new FormatException($"The \"{instrumentation}\" is not recognized as supported metrics instrumentation and cannot be enabled");
-                    }
-                }
-            }
-            else
-            {
-                instrumentations = Enum.GetValues(typeof(MeterInstrumentation))
-                    .Cast<MeterInstrumentation>()
-                    .ToDictionary(
-                        key => Enum.GetName(typeof(MeterInstrumentation), key),
-                        val => val);
-            }
-
-            var disabledInstrumentations = source.GetString(ConfigurationKeys.Metrics.DisabledInstrumentations);
-            if (disabledInstrumentations != null)
-            {
-                foreach (var instrumentation in disabledInstrumentations.Split(Separator))
-                {
-                    instrumentations.Remove(instrumentation);
-                }
-            }
-
-            EnabledInstrumentations = instrumentations.Values.ToList();
+            EnabledInstrumentations = source.ParseEnabledEnumList<MeterInstrumentation>(
+                enabledConfiguration: ConfigurationKeys.Metrics.Instrumentations,
+                disabledConfiguration: ConfigurationKeys.Metrics.DisabledInstrumentations,
+                separator: Separator,
```

Comment:
this is always the same: `,`

Create a const in `ConfigurationSourceExtensions` similarly to `none`

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/939

Reply:
after fix I think we can entirely remove separator parameter.